Sources/BigInt/BigInt.swift

```diff
@@ -0,0 +1,750 @@
+//
+//  BigInt.swift
+//  swift-numerics
+//
+//  Created by Robert Thompson on 2/4/19.
+//
+
+public struct BigInt: BinaryInteger, SignedNumeric, SignedInteger, CustomStringConvertible, LosslessStringConvertible, Hashable {
+    
+    public private(set) var words: Array<UInt>
+
+    public init?(_ description: String) {
+        guard let firstCharacter = description.first else { return nil }
+        
+        var description = description
+        var isNegative: Bool = false
+        if description.count > 1 {
+            if firstCharacter == "-" {
+                isNegative = true
+                description.removeFirst()
+            } else if firstCharacter == "+" {
+                description.removeFirst()
+            }
+        }
```

Comment:
So I tried to find a good way to use Substring instead, but it all looked way uglier than this. Hopefully you're not creating BigInts from Strings in a tight inner loop.

Reply:
@benrimmington, here is an updated version of the initializer. While I would not say that it is simplified, it does have a few advantages of the current version:
1. It does not make an unnecessary copy of the `description` 
2. It removes leading zeros from `description` before the conversion process from `String` to `BigInt` character by character to avoid performing redundant operations.

```swift
public init?(_ description: String) {
    // Ensures `description` is not empty.
    guard let firstCharacter = description.first else { return nil }
    
    var hasPrefixOperator = false
    var isNegative: Bool = false
    
    switch firstCharacter {
    case "-":
        isNegative = true
        fallthrough
    case "+":
        guard description.count > 1 else { return nil }
        hasPrefixOperator = true
    default:
        break
    }
    
    let description = hasPrefixOperator ? description.dropFirst() : description[...]
    
    let digits = Set("0123456789")
    guard description.allSatisfy({ digits.contains($0) }) else { return nil }

    var result: BigInt = 0
    for (i, char) in description.drop(while: { $0 == "0" }).reversed().enumerated() {
        // We're okay to force unwrap here because of the guard above.
        result += BigInt(char.wholeNumberValue!) * pow(10, BigInt(i))
    }
    
    if isNegative {
        result = -result
    }
    
    words = result.words
}
```

Reply:
Hmm, OK, I see, the `description[...]` will give you the whole string as a `Substring` without making a copy anywhere. OK, I like that, I'll change it.

Reply:
I didn't go with exactly what you suggested, I think mine is actually a little cleaner and also handles the case of `description` being empty *before* the prefix is removed, which I somehow missed completely until now!